### PR TITLE
Feat: Automate building of manpages

### LINKED ISSUE #7: Automate manpage building in docker
Or add a CLI option for this while building your docker image (since it takes quite long to build right now)

### PR DESCRIPTION
**Pull request by luarss**
Fixes #7

### CODE REVIEW

File: backend/scrape_docs.py [hunk 1]
```diff
@@ -61,22 +67,85 @@ def scrape_url(url: str, folder_name: str) -> None:
...
+    command = "../../etc/find_messages.py > messages.txt"
+    for module in os.listdir(os.path.join(cur_dir, "OpenROAD/src")):
+        path = os.path.join(cur_dir, "OpenROAD/src", module)
+        if not os.path.isdir(path):
+            continue
+        print("Processing module:", module)
+        os.chdir(path)
+        res = subprocess.run(command, shell=True, capture_output=True)
+        if res.returncode != 0:
+            print(f"Error in finding messages for {module}: {res.stderr}")
```
Comment (luarss) [inline backend/scrape_docs.py:101]:
```suggestion
            print(f"Error in finding messages for {module}: {res.stderr.decode('utf-8')}")
```

File: backend/scrape_docs.py [hunk 2]
```diff
@@ -61,22 +67,85 @@ def scrape_url(url: str, folder_name: str) -> None:
...
+    print("Starting manpages build...")
+
+    # clone repo
+    or_url = "https://github.com/The-OpenROAD-Project/OpenROAD"
+    cur_dir = os.path.dirname(os.path.realpath(__file__))
+    target_dir = os.path.join(cur_dir, "OpenROAD")
+    command = f"git clone {or_url} --depth 1 {target_dir}"
+    res = subprocess.run(command, shell=True, capture_output=True)
+    if res.returncode != 0:
+        print(f"Error in cloning OpenROAD: {res.stderr}")
```
Comment (luarss) [inline backend/scrape_docs.py:80]:
```suggestion
        print(f"Error in cloning OpenROAD: {res.stderr.decode('utf-8')}")
```

File: backend/scrape_docs.py [hunk 3]
```diff
@@ -68,25 +74,88 @@ def scrape_url(url: str, folder_name: str) -> None:
...
+    # clone repo
+    or_url = 'https://github.com/The-OpenROAD-Project/OpenROAD'
+    cur_dir = os.path.dirname(os.path.realpath(__file__))
+    target_dir = os.path.join(cur_dir, 'OpenROAD')
+    command = f"git clone {or_url} --depth 1 {target_dir}"
+    res = subprocess.run(command, shell=True, capture_output=True)
+    if res.returncode != 0:
+        print(f"Error in cloning OpenROAD: {res.stderr.decode('utf-8')}")
+        sys.exit(1)
+    print('Cloned OpenROAD successfully.')
```
Comment (vvbandeira) [inline backend/scrape_docs.py:89]: I would break this into a function.

File: backend/scrape_docs.py [hunk 4]
```diff
@@ -68,25 +74,88 @@ def scrape_url(url: str, folder_name: str) -> None:
...
+        os.chdir(path)
+        res = subprocess.run(command, shell=True, capture_output=True)
+        if res.returncode != 0:
+            print(f"Error in finding messages for {module}: {res.stderr.decode('utf-8')}")
+            continue
+    os.chdir(os.path.join(cur_dir, 'OpenROAD/docs'))
+    num_cores = os.cpu_count()
+    command = f"make clean && make preprocess && make -j{num_cores}"
+    res = subprocess.run(command, shell=True, capture_output=True)
+    print('Finished building manpages.')
```
Comment (vvbandeira) [inline backend/scrape_docs.py:114]: This would be another function

File: backend/scrape_docs.py [hunk 5]
```diff
@@ -68,25 +74,88 @@ def scrape_url(url: str, folder_name: str) -> None:
...
+    for root, _, files in os.walk(dest_dir):
+        for file in files:
+            file_path = os.path.join(root, file)
+            file_name = os.path.basename(file_path)
+            source_dict[file_path] = file_name
+
+    # change back to the file directory
+    os.chdir(cur_dir)
+    shutil.rmtree('OpenROAD')
+    print('Removed OpenROAD temp directory.')
```
Comment (vvbandeira) [inline backend/scrape_docs.py:132]: This would be another function.

**Review by vvbandeira — APPROVED** [review]
Looks good, just some breaking down into functions suggestions. This refactor would be good to add CI steps to test these. I will leave it approved, but if you choose to implement some of the features in this PR just request another review.

**Pull request merged**